test/functional/specs/C35450.js

```diff
@@ -0,0 +1,50 @@
+import { ClientFunction, t } from "testcafe";
+import fixtureFactory from "../helpers/fixtureFactory";
+import createMockVisitor from "../helpers/visitorService/createMockVisitor";
+import createMockOptIn from "../helpers/optIn/createMockOptIn";
+import configureAlloyInstance from "../helpers/configureAlloyInstance";
+import generateId from "../helpers/generateId";
+import {
+  compose,
+  orgMainConfigMain,
+  debugEnabled,
+  migrationEnabled,
+  consentPending
+} from "../helpers/constants/configParts";
+
+fixtureFactory({
+  title:
+    "C35450 - When ID migration is enabled and Visitor and Alloy are both awaiting consent, when consent is given to both, Alloy waits for Visitor to get ECID and then uses this value."
+});
+
+test.meta({
+  ID: "C35448",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const config = compose(
+  orgMainConfigMain,
+  debugEnabled,
+  migrationEnabled,
+  consentPending
+);
+
+const setConsent = ClientFunction(consent => {
+  return window.alloy("setConsent", consent);
+});
+
+const getEcid = ClientFunction(() => {
+  return window.alloy("getIdentity", {}).then(result => {
+    return Promise.resolve(result.ECID);
+  });
+});
+
+test("C35450 - When ID migration is enabled and Visitor and Alloy are both awaiting consent, when consent is given to both, Alloy waits for Visitor to get ECID and then uses this value.", async () => {
+  const ecid = generateId();
+  await createMockVisitor(ecid);
+  await createMockOptIn(true);
```

Comment:
This doesn't match the description.  Visitor isn't awaiting consent.

Reply:
It is pending given that the callback has not been called.

Reply:
The implementation in Alloy is kind of weird because of, in my opinion, some shortcomings of Visitor. When I talked to @wgamboa during implementation, he was saying that if Alloy calls `visitor.getMarketingCloudVisitorID` before opt-in is received, `visitor.getMarketingCloudVisitorID` will return--I think--an empty string. So we have to do this indirection where Alloy interfaces directly with the opt-in object to wait until the user opts in _then_ call `visitor.getMarketingCloudVisitorID`. :/

Reply:
In fact, @liljenback, do you mind adding my comment more or less to https://github.com/adobe/alloy/blob/296563ce509696478d6f33940dc23260b7045026/src/components/Identity/visitorService/getEcidFromVisitorFactory.js#L19 so that later engineers understand this relationship?

Reply:
I think what @jonsnyder might be saying though is that Alloy _technically_ isn't waiting for Visitor to get the ECID. It's more like "When ID migration is enabled and Alloy receives consent, Alloy awaits legacy consent before requesting the ECID from Visitor." I think.

Reply:
Yes, for appMeasurement, when we call getVisitorValues and consent has not been given, it calls the callback with empty values.  We should duplicate that behavior in the mocks here.  This does bring up an interesting question though on whether we should be including the real Visitor.js on the page for these integration tests.

Reply:
I was thinking about including Visitor, but it would be nice if we could avoid it. We're only using the functionality to get the ECID.

Reply:
I was also wondering why we weren't using a real Visitor instance. It would give us the most accurate results. What's your reasoning for wanting to avoid it even if it's just being used for ECID @liljenback?

Reply:
Well we call getMarketingCloudVisitorID with a callback and that resolves with an ID. The Visitor library has a lot more functionality and including that overhead didn't seem necessary.

Reply:
@liljenback @Aaronius @jonsnyder 

We already decided to use the real Visitor.js back when we planned this test, mainly because it's an E2E test. Meaning when Visitor changes, we'd love to to know if it's still compatible with Alloy. So let's use the `latest` tag from the Github repo where Visitor gets released. I believe it's this guy:

https://github.com/Adobe-Marketing-Cloud/id-service/releases/latest

Thanks.